### ISSUE TITLE
Request: callback on dialog.Dialog completion

### ISSUE THREAD

**Issue by MagicalTux** [fyne-io/fyne#1540]
### Is your feature request related to a problem? Please describe:
`dialog.Dialog` could have a `SetOnClosed(callback func())` method.

The goal in my case would be to restore focus to the right widget on the parent window, but I'm guessing other people could have other goals.

### Is it possible to construct a solution with the existing API?
No

### Describe the solution you'd like to see:

add `SetOnClosed(callback func())` to `dialog.Dialog`

**Comment by MagicalTux** [comment]
Actually Dialog do have SetOnClosed. Nevermind.

**Comment by MagicalTux** [comment]
Did some testing, `SetOnClosed()` isn't triggered by the OK button.

Having the ability to set a callback when this is closed would be great.

![image](https://user-images.githubusercontent.com/333658/99162528-ffa08700-2741-11eb-96b9-9a84a51a871c.png)

**Comment by AlbinoGeek** [comment]
Don't all dialogs have a built-in callback in their constructor, which covers all cases except `OnClosed` already?

**Comment by andydotxyz** [comment]
Can you please share the code where `SetOnClosed()` does not work? It should trigger for every dialog irrespective of how the user dismissed it.

**Comment by MagicalTux** [comment]
The code:

```go

		log.Printf("creating dialog")
		d := dialog.NewError(errors.New(msg), l.w)

		d.SetOnClosed(func() {
			log.Printf("onClosed called")
			// restore focus
			l.restoreFocus()
		})
```

Resulting output:
```
2020/11/15 21:11:48 creating dialog
```

Clicking OK doesn't trigger the "onClosed called" log as one would expect

**Comment by MagicalTux** [comment]
(note, I use this way of creating the error dialog because I want the nice error icon but it's a pain to generate as customdialog doesn't allow setting the icon)

**Comment by andydotxyz** [comment]
Ah, I figured this out - sorry it relates to an age-old bug that we could not fix until 2.0.
You forgot to call `d.Show()`.
The bug is that it should not have been shown until Show was called - but by mistake New showed dialogs many versions ago.
In 2.0 this will be resolved but as it's a breaking behaviour change we could not fix before.

**Comment by andydotxyz** [comment]
Closing as duplicate as it's a known issue, will be resolved soon on the `develop` branch as we start working towards 2.0
